### PR TITLE
Divide ranking map state from ranking state

### PR DESCRIPTION
**Pull request by ipdae**
- Divide ranking map state from ranking state

### CODE REVIEW

File: Lib9c/Action/CreateAvatar.cs [hunk 1]
```diff
@@ -147,15 +152,15 @@ public override IAccountStateDelta Execute(IActionContext context)
...
 
-        private static AvatarState CreateAvatarState(
-            string name,
+        private static AvatarState CreateAvatarState(string name,
             Address avatarAddress,
             IActionContext ctx,
-            MaterialItemSheet materialItemSheet
-        )
+            MaterialItemSheet materialItemSheet,
+            Address mapAddress)
```
Comment (boscohyun) [inline Lib9c/Action/CreateAvatar.cs:163]:
```suggestion
            Address rankingMapAddress)
```
조금 더 풀어 써주는 게 어떤가요?

**Review by boscohyun — APPROVED** [review]

File: Lib9c/Model/State/AvatarState.cs [hunk 2]
```diff
@@ -50,6 +50,8 @@ public class AvatarState : State, ICloneable
         public string NameWithHash { get; private set; }
         public int Nonce { get; private set; }
 
+        public readonly Address rankingMapAddress;
```
Comment (limebell) [inline Lib9c/Model/State/AvatarState.cs:53]:
```suggestion
        public readonly Address RankingMapAddress;
```

File: Lib9c/Model/State/AvatarState.cs [hunk 3]
```diff
@@ -106,6 +109,8 @@ public static Address CreateAvatarAddress()
                 );
                 combinationSlotAddresses.Add(slotAddress);
             }
+
+            this.rankingMapAddress = rankingMapAddress;
```
Comment (limebell) [inline Lib9c/Model/State/AvatarState.cs:113]:
```suggestion
            RankingMapAddress = rankingMapAddress;
```

File: Lib9c/Model/State/RankingState.cs [hunk 4]
```diff
@@ -1,152 +1,68 @@
...
-    /// </summary>
     [Serializable]
     public class RankingState : State
     {
         public static readonly Address Address = Addresses.Ranking;
-        private readonly Dictionary<Address, RankingInfo> _map;
-
-        public List<RankingInfo> OrderedRankingInfos { get; private set; }
+        public const int RankingMapCapacity = 100;
+        public readonly Dictionary<Address, ImmutableHashSet<Address>> rankingMap;
```
Comment (limebell) [inline Lib9c/Model/State/RankingState.cs:16]:
```suggestion
        public readonly Dictionary<Address, ImmutableHashSet<Address>> RankingMap;
```

**Review by limebell — COMMENTED** [review]
For consistency :)

**Review by Unengine — APPROVED** [review]

File: Lib9c/Action/InvalidAddressException.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Nekoyume.Action
+{
+    public class InvalidAddressException : InvalidOperationException
```
Comment (longfin) [inline Lib9c/Action/InvalidAddressException.cs:6]:
```suggestion
    [Serializable]
    public class InvalidAddressException : InvalidOperationException
```

File: Lib9c/Action/RankingExceededException.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Nekoyume.Action
+{
+    public class RankingExceededException: InvalidOperationException
```
Comment (longfin) [inline Lib9c/Action/RankingExceededException.cs:6]:
```suggestion
    [Serializable]
    public class RankingExceededException: InvalidOperationException
```

**Review by longfin — APPROVED** [review]

**Review by longfin — APPROVED** [review]

**Pull request merged**